Framework/Core/src/LifetimeHelpers.cxx

```diff
@@ -421,6 +422,11 @@ ExpirationHandler::Handler LifetimeHelpers::enumerate(ConcreteDataMatcher const&
     dh.payloadSize = sizeof(counter_t);
     dh.payloadSerializationMethod = gSerializationMethodNone;
     dh.tfCounter = timestamp;
+    try {
+      dh.runNumber = atoi(services.get<DataTakingContext>().runNumber.c_str());
+    } catch (...) {
+      dh.runNumber = 0;
```

Comment:
```suggestion
      dh.runNumber = -1;
```
@davidrohr something like this?

Reply:
Probably I would rather reserve a range of ~100 invalid positive numbers with meaning, like we have a range for unanchored MC. You can discuss the range with @ehellbar and RC.
Then, the invalid runNumber check would also need to check for that range. And if we get an error with a number from that range, it is clear how it happened.